admin/schema.php

```diff
@@ -884,6 +884,27 @@
 
 # Release marker: 1.3.0
 
+# Count / occurrences / votes 
+$g_upgrade[210] = array( 'AddColumnSQL', array( db_get_table( 'bug' ), "
+	votes					I		NOTNULL DEFAULT '1'",
+	) );
+$g_upgrade[211] = array( 'CreateIndexSQL', array( 'idx_bug_count', db_get_table( 'bug' ), 'votes' ) );	
+
+# A standalone table is required for "alias", because not all DB allow multiple NULLs in UNIQUE index
+$g_upgrade[212] = array('CreateTableSQL', array( db_get_table( 'bug_alias' ), "
+	bug_id					I		UNSIGNED NOTNULL PRIMARY,
+	project_id				I		UNSIGNED NOTNULL,
+	alias					C(128)	$t_notnull",
+	$t_table_options
+	) );
+$g_upgrade[213] = array( 'CreateIndexSQL', array( 'idx_bug_alias_project', db_get_table( 'bug_alias' ), 'project_id' ) );	
+$g_upgrade[214] = array( 'CreateIndexSQL', array( 'idx_bug_alias_alias', db_get_table( 'bug_alias' ), 'alias' ) );	
+$g_upgrade[215] = array( 'CreateIndexSQL', array( 'idx_bug_alias_uniq', db_get_table( 'bug_alias' ), 'project_id,alias', array( 'UNIQUE' ) ) );
+
+# Add API token kind, allowing to have different types of API tokens
+$g_upgrade[216] = array( 'AddColumnSQL', array( db_get_table( 'api_token' ), "
+	kind					I		NOTNULL DEFAULT '0'",
```

Comment:
Instead of having types, I would rather add the concept of scopes to tokens.  For example, default can be "['All']', and in the case of reporting {'Issue.Report'}.  SOAP API and web app would check the scope to determine if a user should be able to login, report an issue, etc.  Hence, the tokens created with "['Issue.Report']" would only be able to report issues, and no more.  The format of this would be a json array of strings.